src/client/metadataApiRetrieve.ts

```diff
@@ -116,7 +116,10 @@ export class RetrieveResult implements MetadataTransferResult {
     // Add file responses for components that support partial delete (e.g., DigitalExperience)
     // where pieces of the component were deleted in the org, then retrieved.
     while (partialDeleteFileResponses.length) {
-      this.fileResponses.push(partialDeleteFileResponses.pop());
```

Comment:
I think this code was fine, but TS wasn't sure that the result of pop would always be defined